pkg/resources/aws/aws.go

```diff
@@ -2078,11 +2078,17 @@ func ListIAMOIDCProviders(cloud fi.Cloud, vpcID, clusterName string) ([]*resourc
 			descReq := &iam.GetOpenIDConnectProviderInput{
 				OpenIDConnectProviderArn: arn,
 			}
-			resp, err := c.IAM().GetOpenIDConnectProvider(descReq)
-			if err != nil && awsup.AWSErrorCode(err) == iam.ErrCodeNoSuchEntityException {
-				continue
-			} else if err != nil {
-				return nil, fmt.Errorf("error getting IAM OIDC Provider: %v", err)
+			resp, err := c.IAM().GetOpenIDConnectProvider(ctx, descReq)
+			if err != nil {
+				var nse *iamtypes.NoSuchEntityException
+				if errors.As(err, &nse) {
+					klog.Warningf("could not find IAM OIDC Provider %q. Resource may already have been deleted: %v", aws.StringValue(arn), nse)
+					continue
+				} else if awserror, ok := err.(smithy.APIError); ok && awserror.ErrorCode() == "403" {
```

Comment:
This feels like a more fragile API than we had before.  Maybe we should hide `if awserror, ok := err.(smithy.APIError); ok && awserror.ErrorCode() == "403" ` behind a function like `IsHTTP403Error(err error) bool` ?